src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/lib/pipeline.js

```diff
@@ -108,24 +107,44 @@ export default class Pipeline {
     return result;
   }
 
-  // Updates the state of the pipeline and processors with the results
-  // from an ingest simulate call.
-  applySimulateResults(results) {
-    //update the outputObject of each processor
-    results.forEach((result) => {
+  _updateProcessorOutputs(simulateResults) {
+    simulateResults.forEach((result) => {
       const processor = this.getProcessorById(result.processorId);
 
       processor.outputObject = _.get(result, 'output');
       processor.error = _.get(result, 'error');
     });
+  }
 
-    //update the inputObject of each processor
-    results.forEach((result) => {
-      const processor = this.getProcessorById(result.processorId);
+  //Updates the error state of the pipeline and its processors
+  //If a pipeline compile error is returned, lock all processors but the error
+  //If a pipeline data error is returned, lock all processors after the error
+  _updateErrorState() {
```

Comment:
If these methods that start with an `_` are supposed to be private, don't just rely on a naming scheme, hide them in the closure so that outside code cannot access them.